util/hack/hack.go

```diff
@@ -24,8 +24,15 @@ type MutableString string
 // String converts slice to MutableString without copy.
 // The MutableString can be converts to string without copy.
 // Use it at your own risk.
-func String(b []byte) MutableString {
-	return *(*MutableString)(unsafe.Pointer(&b))
+func String(b []byte) (s MutableString) {
```

Comment:
Can you provide a [benchmark](https://golang.org/pkg/testing/#hdr-Benchmarks) for this function and provide comparable results before and after.